src/realm.js

```diff
@@ -564,17 +569,29 @@ export class Realm {
         let tval = gen.derive(value.types, value.values, [value], ([n]) => n, {
           skipInvariant: true,
         });
+        tval.mightBeEmpty = value.mightBeEmpty;
         tvalFor.set(key, tval);
       }
     });
     pbindings.forEach((val, key, map) => {
       let path = key.pathNode;
       let tval = tvalFor.get(key);
       invariant(tval !== undefined);
+      let mightBeEmpty = tval.mightBeEmpty;
       gen.emitStatement([key.object, tval], ([o, v]) => {
         invariant(path !== undefined);
         let lh = path.buildNode([o]);
-        return t.expressionStatement(t.assignmentExpression("=", (lh: any), v));
+        let r = t.expressionStatement(t.assignmentExpression("=", (lh: any), v));
```

Comment:
The logic looks good to me. But I do not like the fact that we are stuffing more generator serialization code into realm.js. It would be more centralized if we put all generator related code as member functions of generator class.

Reply:
I'm not inclined to agree with this sentiment. Centralizing all Babel AST generation in the generator does not make the code more readable or maintainable. Moreover, the architecture specifically provides build functions so that this logic can be distributed to places where they can be read in context.